pandas/core/groupby/ops.py

```diff
@@ -405,7 +406,7 @@ def _reconstruct_ea_result(
         """
         dtype: BaseMaskedDtype | StringDtype
 
-        if isinstance(values.dtype, StringDtype):
+        if is_string_dtype(values.dtype):
```

Comment:
doesnt make a difference here bc we know we have EA, but we should have a func like is_string_dtype that excludes object